django/contrib/admin/tests.py

```diff
@@ -26,18 +57,37 @@ class AdminSeleniumWebDriverTestCase(StaticLiveServerTestCase):
         'django.contrib.sessions',
         'django.contrib.sites',
     ]
-    webdriver_class = 'selenium.webdriver.firefox.webdriver.WebDriver'
+
+    @classmethod
+    def _get_local_webdriver_class(self, spec):
+        """
+        Return the webdriver for a local browser corresponding to the given
+        browser spec.
+        """
+        browsers = {
+            'firefox': 'selenium.webdriver.Firefox',
+            'safari': 'selenium.webdriver.Safari',
+            'opera': 'selenium.webdriver.Opera',
+            'ie': 'selenium.webdriver.Ie',
+            'chrome': 'selenium.webdriver.Chrome',
+            'phantomjs': 'selenium.webdriver.PhantomJS',
+            'edge': 'selenium.webdriver.Edge',
```

Comment:
Did you check if there's a way to pull this mapping from selenium or build it automatically so that if browsers are added there, we don't have to update Django?